File: dbpro/src/main/java/kr/ac/sejong/feat/advertisement/MovieAdDAO.java
package kr.ac.sejong.feat.advertisement;

import kr.ac.sejong.config.JDBCManager;

import java.util.List;

public class MovieAdDAO {

	private MovieAdDAO() {
	}

	private static MovieAdDAO instance;

	public static MovieAdDAO getInstance() {
		if (instance == null) {
			instance = new MovieAdDAO();
		}
		return instance;
	}

	public List<MovieAdDTO> selectMovieADs() {
		final String query = "SELECT " +
			"	CNT, " +
			"	MOVIE_CODE, " +
			"	AD_TITLE " +
			"FROM " +
			"	MOVIE_AD";

		return JDBCManager
			.getInstance()
			.queryForList(query, MovieAdDTO.class);
	}

}

File: dbpro/src/main/java/kr/ac/sejong/feat/advertisement/MovieAdDAO.kt
package kr.ac.sejong.feat.advertisement

import kr.ac.sejong.config.JDBCManager

class MovieAdDAO private constructor() {
    companion object {
        @JvmStatic
        var instance: MovieAdDAO? = null
            get() {
                if (field == null) {
                    field = MovieAdDAO()
                }
                return field
            }
            private set
    }

    fun selectMovieADs(): List<MovieAdDTO> {
        val query = """
            SELECT
                CNT,
                MOVIE_CODE,
                AD_TITLE
            FROM
                MOVIE_AD
        """

        return JDBCManager
            .getInstance()
            .queryForList(query, MovieAdDTO::class.java)
    }
}